packages/@angular/cli/models/webpack-configs/typescript.ts

```diff
@@ -19,15 +19,16 @@ function _createAotPlugin(wco: WebpackConfigOptions, options: any) {
   let hostOverrideFileSystem: any = {};
   // process environment file replacement
   if (appConfig.environments) {
-    if (!('source' in appConfig.environments)) {
-      throw new SilentError(`Environment configuration does not contain "source" entry.`);
+    if (!appConfig.environmentSource) {
```

Comment:
Please verify if `environment.source` is set here and explain to the user the change s/he needs to make.